lib/oli/resources/collaboration.ex

```diff
@@ -1306,14 +1315,23 @@ defmodule Oli.Resources.Collaboration do
   resource posts, the annotated block id is nil.
   """
   def list_post_counts_for_user_in_section(section_id, resource_id, user_id, visibility) do
+    filter_by_visibility =
+      case visibility do
+        :private ->
+          dynamic([p], p.visibility == ^visibility and p.user_id == ^user_id)
+
+        _ ->
+          dynamic([p], p.visibility == ^visibility)
+      end
```

Comment:
We should implement this same filter clause in `search_posts_for_user_in_point_block/6` to prevent a student from searching through other students' private notes.

https://github.com/Simon-Initiative/oli-torus/assets/74839302/ef7a28db-83ad-46b2-a089-94d43af4becf

Reply:
Good catch. Fixed